scripts/tcl_commands/all.tcl

```diff
@@ -597,40 +597,56 @@ proc save_views {args} {
-    if { $::env(DIODE_INSERTION_STRATEGY) == 2 } {
-        set report_file [open $::env(magic_report_file_tag).antenna_violators.rpt r]
-        set violators [split [string trim [read $report_file]]]
-        close $report_file
-        # may need to speed this up for extremely huge files using hash tables
-        exec echo $violators >> $::env(TMP_DIR)/vios.txt
-        #foreach violator $violators {
-        #	if { 1 } {; #  VERBOSE
-        #		puts_info "Healing $violator"
-        #	}
-        # try_catch $::env(SCRIPTS_DIR)/heal_antenna_violators.sh $violator
-        #	try_catch sed -i -E "/COMPONENTS/,/END COMPONENTS/ s/(- ANTENNA_${violator}\\S+) $::env(FAKEDIODE_CELL)/\\1 $::env(DIODE_CELL)/" $::env(tritonRoute_result_file_tag).def
-        #}
-        #try_catch sed -i -E "/COMPONENTS/,/END COMPONENTS/ s/(- ANTENNA_${violator}\\S+) $::env(FAKEDIODE_CELL)/\\1 $::env(DIODE_CELL)/g" $::env(tritonRoute_result_file_tag).def
-        #try_catch sed -f <(printf '/COMPONENTS/,/END COMPONENTS/ s/ANTENNA_%s\\S+ $::env(FAKEDIODE_CELL)/\\1ANTENNA_%s\\S+ $::env(DIODE_CELL)/g\n' $(<$::env(TMP_DIR)/vios.txt)) <($::env(tritonRoute_result_file_tag).def) >$::env(tritonRoute_result_file_tag).def
-        #try_catch sed -f <(printf '/COMPONENTS/,/END COMPONENTS/ s/(- ANTENNA_%s\\S+)$::env(FAKEDIODE_CELL)/\\1 $::env(DIODE_CELL)/g/\n' $(<$::env(TMP_DIR)/vios.txt)) $::env(tritonRoute_result_file_tag).def
-        #exec sed -f <(printf '/COMPONENTS/,/END COMPONENTS/ /(- ANTENNA_%s\\S+)/ s/$::env(FAKEDIODE_CELL)/$::env(DIODE_CELL)/g\n' (<$::env(TMP_DIR)/vios.txt)) $::env(tritonRoute_result_file_tag).def
-        #foreach violator $violators {
-        #	if { 1 } {;
-        #		puts_info "Healing $violator"
-        #	}
-        #	set tmpString "'/COMPONENTS/,/END COMPONENTS/ s/(- ANTENNA_${violator}\\S+) $::env(FAKEDIODE_CELL)/\\1 $::env(DIODE_CELL)/'"
-        # try_catch $::env(SCRIPTS_DIR)/heal_antenna_violators.sh $violator
-        #	try_catch sed -i -E $tmpString $::env(tritonRoute_result_file_tag).def
-        #}
-        #exec sed -i '' -e s/Red/$color1/g -e s/Blue/$color2/g {} \;
-        exec python3 $::env(SCRIPTS_DIR)/fakeDiodeReplace.py -v $::env(TMP_DIR)/vios.txt -d $::env(tritonRoute_result_file_tag).def -f $::env(FAKEDIODE_CELL) -t $::env(DIODE_CELL)
-        puts_info "DONE HEALING ANTENNA VIOLATORS"
-
-    }
+	# requires a pre-existing report containing a list of cells (-pins?)
+	# that need the real diode in place of the fake diode:
+	# $::env(magic_tmp_file_tag).antenna_violators.rpt
+	# => fixes the routed def
+	if { $::env(DIODE_INSERTION_STRATEGY) == 2 } {
+		#Legacy from magic antenna checker
+		#set report_file [open $::env(magic_report_file_tag).antenna_violators.rpt r]
+		#	set violators [split [string trim [read $report_file]]]
+		#close $report_file
+		# may need to speed this up for extremely huge files using hash tables
+		#exec echo $violators >> $::env(TMP_DIR)/vios.txt
+		
+		
+		# Shell solution for replacement
```

Comment:
This really doesn't need to be kept around here, does it? It can live either in the git history or in the new python scripts as a reference implementation

Reply:
I agree. I guess a little clean-up wouldn't hurt and will make the block "less" confusing.